src/parser.js

```diff
@@ -2101,27 +2141,52 @@ export class GenericParser extends Tokenizer {
 
   parseObjectExpression() {
     let startState = this.startNode();
-
     this.lex();
-
+    let rest = null;
     let properties = [];
     while (!this.match(TokenType.RBRACE)) {
-      let property = this.inheritCoverGrammar(this.parsePropertyDefinition);
-      properties.push(property);
-      if (!this.match(TokenType.RBRACE)) {
+      let property = null;
+      let isSpreadProperty = false;
+      let spreadPropertyOrAssignmentTarget = null;
+      if (this.match(TokenType.ELLIPSIS)) {
+        spreadPropertyOrAssignmentTarget = this.parseSpreadPropertyDefinition();
+        isSpreadProperty = true;
+
+        property = spreadPropertyOrAssignmentTarget.expression;
+        if (property.type === 'ObjectExpression' || property.type === 'ArrayExpression') {
```

Comment:
For consistency with `parsePropertyDefinition`, this logic should be in `parseSpreadPropertyDefinition`.